### PR TITLE
feat: add test for build-args

### PR DESCRIPTION
**Pull request by laurentsenta**
example output:
https://github.com/laurentsenta/docker-container-action/actions/runs/6034742092

### CODE REVIEW

**Comment by laurentsenta** [comment]
@galargh the experiment I was talking about,
the container does output:
```
arg-1=hello
arg-2=foo
arg-3=world
```

Which is pipe'd to the GITHUB_OUTPUT in theory, but my outputs are empty.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: Test
+on:
+  push:
```
Comment (galargh) [inline .github/workflows/test.yml:3]:
```suggestion
  push:
    branches:
      - main
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  pull_request:
+jobs:
+  test-build-args:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: github
+        # todo: fix this path
+        uses: pl-strflt/docker-container-action/.github/actions/github@main
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
        uses: ./.github/actions/github
```

File: tests/test-build-args.Dockerfile
Comment (galargh) [inline tests/test-build-args.Dockerfile:1]: Can we rename to `Dockerfile.test-build-args`?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      - id: debug
+        run: |
+          echo "GITHUB OUTPUT: $GITHUB_OUTPUT"
+          echo "PWD: $PWD"
+          echo `ls -la`
+          cat tests/test-build-args.Dockerfile
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
          cat tests/Dockerfile.test-build-args
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      - name: build the container and run it
+        id: run
+        # todo: fix this path
+        uses: pl-strflt/docker-container-action@main
+        with:
+          repository: ${{ steps.github.outputs.action_repository }}
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
          repository: ${{ github.repository }}
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          ref: ${{ steps.github.outputs.action_sha || steps.github.outputs.action_ref }}
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
          ref: ${{ github.ref }}
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+          dockerfile: tests/test-build-args.Dockerfile
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
          dockerfile: tests/Dockerfile.test-build-args
```

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+          args: --arg1 --arg2 --arg3
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
```

File: tests/test-build-args.Dockerfile [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN echo "#!/bin/sh" > /script.sh
+RUN echo "echo \"arg-1=$ARG_1\" | tee /output" >> /script.sh
+RUN echo "echo \"arg-2=$ARG_2\" | tee /output" >> /script.sh
+RUN echo "echo \"arg-3=$ARG_3\" | tee /output" >> /script.sh
+# also echo the shell args
+RUN echo "for arg in \"\$@\"" >> /script.sh
+RUN echo "do" >> /script.sh
+RUN echo "  echo \"arg: \$arg\"" >> /script.sh
+RUN echo "done" >> /script.sh
```
Comment (galargh) [inline tests/test-build-args.Dockerfile]:
```suggestion
```

File: tests/test-build-args.Dockerfile [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+RUN chmod +x /script.sh
+
+ENTRYPOINT ["/script.sh"]
```
Comment (galargh) [inline tests/test-build-args.Dockerfile]:
```suggestion
ENTRYPOINT ["/script.sh"]

```

File: .github/workflows/test.yml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+            ARG_1:hello
+            ARG_2:foo
+            ARG_3:world
+          opts: --volume ${GITHUB_OUPUT}:/output
+      - name: check the output
+        env:
+          OUTPUTS: ${{ steps.run.outputs.outputs }}
+          # ARG_1: ${{ steps.run.outputs.outputs.arg-1 }}
+          # ARG_2: ${{ steps.run.outputs.outputs.arg-2 }}
+          # ARG_3: ${{ steps.run.outputs.outputs.arg-3 }}
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
          OUTPUTS: ${{ toJSON(steps.run.outputs) }}
          ARG_1: ${{ steps.run.outputs.arg-1 }}
          ARG_2: ${{ steps.run.outputs.arg-2 }}
          ARG_3: ${{ steps.run.outputs.arg-3 }}
```

File: .github/workflows/test.yml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+          echo "PWD: $PWD"
+          echo `ls -la`
+          cat tests/test-build-args.Dockerfile
+      - name: build the container and run it
+        id: run
+        # todo: fix this path
+        uses: pl-strflt/docker-container-action@main
+        with:
+          repository: ${{ github.repository }}
+          ref: ${{ github.ref }}
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
          ref: ${{ github.sha }}
```

File: .github/workflows/test.yml [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+      - id: debug
+        run: |
+          echo "GITHUB OUTPUT: $GITHUB_OUTPUT"
+          echo "PWD: $PWD"
+          echo `ls -la`
+          cat tests/test-build-args.Dockerfile
+      - name: build the container and run it
+        id: run
+        # todo: fix this path
+        uses: pl-strflt/docker-container-action@main
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
        uses: ./.
```

File: .github/workflows/test.yml [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+      - id: debug
+        run: |
+          echo "GITHUB OUTPUT: $GITHUB_OUTPUT"
+          echo "PWD: $PWD"
+          echo `ls -la`
+          cat tests/test-build-args.Dockerfile
+      - name: build the container and run it
+        id: run
+        # todo: fix this path
+        uses: ./.
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
        uses: ./
```

File: .github/workflows/test.yml [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+      - uses: actions/checkout@v3
+      - id: github
+        # todo: fix this path
+        uses: ./.github/actions/github
+      - id: debug
+        run: |
+          echo "GITHUB OUTPUT: $GITHUB_OUTPUT"
+          echo "PWD: $PWD"
+          echo `ls -la`
+          cat tests/Dockerfile.test-build-args
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
```

File: .github/workflows/test.yml [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+  pull_request:
+jobs:
+  test-build-args:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: github
+        # todo: fix this path
+        uses: ./.github/actions/github
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
```

File: .github/workflows/test.yml [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+        uses: ./
+        with:
+          repository: ${{ github.repository }}
+          ref: ${{ github.sha }}
+          dockerfile: tests/Dockerfile.test-build-args
+          build-args: |
+            ARG_1:hello
+            ARG_2:foo
+            ARG_3:world
+          opts: --volume ${GITHUB_OUPUT}:/output
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
```

File: .github/workflows/test.yml [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+      - name: check the output
+        env:
+          OUTPUTS: ${{ toJSON(steps.run.outputs) }}
+          ARG_1: ${{ steps.run.outputs.arg-1 }}
+          ARG_2: ${{ steps.run.outputs.arg-2 }}
+          ARG_3: ${{ steps.run.outputs.arg-3 }}
```
Comment (galargh) [inline .github/workflows/test.yml]:
```suggestion
          OUTPUTS: ${{steps.run.outputs.outputs }}
          ARG_1: ${{ fromJSON(steps.run.outputs.outputs).arg-1 }}
          ARG_2: ${{ fromJSON(steps.run.outputs.outputs).arg-2 }}
          ARG_3: ${{ fromJSON(steps.run.outputs.outputs).arg-3 }}
```

File: tests/Dockerfile.test-build-args [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+FROM alpine:3.12.0
+
+ARG ARG_1=default_1
+ARG ARG_2=default_2
+ARG ARG_3=default_3
+
+RUN echo "#!/bin/sh" > /script.sh
+RUN echo "echo \"arg-1=$ARG_1\" | tee /output" >> /script.sh
+RUN echo "echo \"arg-2=$ARG_2\" | tee /output" >> /script.sh
+RUN echo "echo \"arg-3=$ARG_3\" | tee /output" >> /script.sh
```
Comment (galargh) [inline tests/Dockerfile.test-build-args]:
```suggestion
RUN echo 'OUTPUT="${GITHUB_OUTPUT/"/home/runner/work/_temp/_runner_file_commands"/"/github/file_commands"}"' >> /script.sh
RUN echo "echo \"arg-1=$ARG_1\" | tee \$OUTPUT" >> /script.sh
RUN echo "echo \"arg-2=$ARG_2\" | tee \$OUTPUT" >> /script.sh
RUN echo "echo \"arg-3=$ARG_3\" | tee \$OUTPUT" >> /script.sh
```

File: tests/Dockerfile.test-build-args [hunk 19]
```diff
@@ -0,0 +1,16 @@
+# a simple container that takes 3 args and generate a shell script that prints them
+FROM alpine:3.12.0
+
+ARG ARG_1=default_1
+ARG ARG_2=default_2
+ARG ARG_3=default_3
+
+RUN echo "#!/bin/sh" > /script.sh
+RUN echo 'OUTPUT="${GITHUB_OUTPUT/"/home/runner/work/_temp/_runner_file_commands"/"/github/file_commands"}"' >> /script.sh
```
Comment (galargh) [inline tests/Dockerfile.test-build-args]:
```suggestion
RUN echo 'GITHUB_FILE_COMMANDS="$(dirname "$GITHUB_OUTPUT")" >> /script.sh
RUN echo 'OUTPUT="${GITHUB_OUTPUT/"$FILE_COMMANDS"/"/github/file_commands"}"' >> /script.sh
```

File: tests/Dockerfile.test-build-args [hunk 20]
```diff
@@ -0,0 +1,17 @@
+# a simple container that takes 3 args and generate a shell script that prints them
+FROM alpine:3.12.0
+
+ARG ARG_1=default_1
+ARG ARG_2=default_2
+ARG ARG_3=default_3
+
+RUN echo "#!/bin/sh" > /script.sh
+RUN echo 'GITHUB_FILE_COMMANDS="$(dirname "$GITHUB_OUTPUT")" >> /script.sh
+RUN echo 'OUTPUT="${GITHUB_OUTPUT/"$FILE_COMMANDS"/"/github/file_commands"}"' >> /script.sh
```
Comment (galargh) [inline tests/Dockerfile.test-build-args]:
```suggestion
RUN echo 'GITHUB_FILE_COMMANDS="$(dirname "$GITHUB_OUTPUT")"' >> /script.sh
RUN echo 'OUTPUT="${GITHUB_OUTPUT/"$GITHUB_FILE_COMMANDS"/"/github/file_commands"}"' >> /script.sh
```

File: tests/Dockerfile.test-build-args [hunk 21]
```diff
@@ -0,0 +1,17 @@
...
+ARG ARG_1=default_1
+ARG ARG_2=default_2
+ARG ARG_3=default_3
+
+RUN echo "#!/bin/sh" > /script.sh
+RUN echo 'GITHUB_FILE_COMMANDS="$(dirname "$GITHUB_OUTPUT")"' >> /script.sh
+RUN echo 'OUTPUT="${GITHUB_OUTPUT/"$GITHUB_FILE_COMMANDS"/"/github/file_commands"}"' >> /script.sh
+RUN echo "echo \"arg-1=$ARG_1\" | tee \$OUTPUT" >> /script.sh
+RUN echo "echo \"arg-2=$ARG_2\" | tee \$OUTPUT" >> /script.sh
+RUN echo "echo \"arg-3=$ARG_3\" | tee \$OUTPUT" >> /script.sh
```
Comment (galargh) [inline tests/Dockerfile.test-build-args]:
```suggestion
RUN echo "echo \"arg-1=$ARG_1\" | tee -a \$OUTPUT" >> /script.sh
RUN echo "echo \"arg-2=$ARG_2\" | tee -a \$OUTPUT" >> /script.sh
RUN echo "echo \"arg-3=$ARG_3\" | tee -a \$OUTPUT" >> /script.sh
```

**Comment by galargh** [comment]
@laurentsenta It works now :) The main issue was that GITHUB_OUTPUT of https://github.com/pl-strflt/docker-container-action/blob/main/action.yml#L92 is not the same as GITHUB_OUTPUT of https://github.com/pl-strflt/docker-container-action/blob/3eb52301abbc79590ec62e79a0eaea90ea08e575/.github/workflows/test.yml#L13. Have a look at the diff for all changes.

File: .github/workflows/test.yml [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+  pull_request:
+jobs:
+  test-build-args:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: build the container and run it
+        id: run
+        # todo: fix this path
```
Comment (laurentsenta) [inline .github/workflows/test.yml]:
```suggestion
```

File: tests/Dockerfile.test-build-args [hunk 23]
```diff
@@ -0,0 +1,17 @@
+# a simple container that takes 3 args and generate a shell script that prints them
+FROM alpine:3.12.0
+
+ARG ARG_1=default_1
+ARG ARG_2=default_2
+ARG ARG_3=default_3
+
+RUN echo "#!/bin/sh" > /script.sh
+RUN echo 'GITHUB_FILE_COMMANDS="$(dirname "$GITHUB_OUTPUT")"' >> /script.sh
+RUN echo 'OUTPUT="${GITHUB_OUTPUT/"$GITHUB_FILE_COMMANDS"/"/github/file_commands"}"' >> /script.sh
```
Comment (laurentsenta) [inline tests/Dockerfile.test-build-args:10]: @galargh how did you came up with this path & transformation?

Comment (galargh) [reply tests/Dockerfile.test-build-args:10]: 1. I looked at what volumes are set up when we start the container.

Screenshot 2023-08-31 at 10 17 31

2. I looked at the env on a default runner.

3. I noticed that the root dir for `GITHUB_OUTPUT` is `GITHUB_FILE_COMMANDS`.

4. I drew the conclusion that `GITHUB_FILE_COMMANDS` has to be replaced with `/github/file_commands` (as per docker container volume) in `GITHUB_OUTPUT`.

**Pull request merged**